apps/settings/js/panels/addons/addons_list.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * UI for Addons panel's functionality.
+ *
+ * @module AddonsList
+ */
+define(function(require) {
+  'use strict';
+
+  var ListView = require('modules/mvvm/list_view');
+  var addonItemTemplate = require('panels/addons/addons_template');
+
+  function AddonsList(root, manager) {
+    this._enabled = false;
+    this._listRoot = root;
+    this._addonsManager = manager;
+    this._listView = ListView(this._listRoot,
+      this._addonsManager.addons.apps, addonItemTemplate);
+  }
+
+  AddonsList.prototype = {
+    set enabled(value) {
+      if (value !== this._enabled) {
+        this._listView.enabled = this._enabled = value;
+      }
+    }
+  };
+
+  return AddonsList;
```

Comment:
We don't return prototype directly in setting app. Please return a function that return the instance of the prototype. This give us a chance to take additional parameters and return different objects when necessary.